src/main/java/InfrastructureManager/ScenarioDispatcher.java

```diff
@@ -0,0 +1,61 @@
+package InfrastructureManager;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.io.IOException;
+
+public class ScenarioDispatcher implements MasterOutput {
+    private Scenario scenario;
+
+    public ScenarioDispatcher() {
+        this.scenario = null;
+    }
+    @Override
+    public void out(String response) {
+        String[] command = response.split(" ");
+        switch (command[0]) {
+            case "fromFile":
+                scenarioFromFile(command[1]);
+                break;
+            case "run" :
+                runScenario();
+                break;
+            case "pause" :
+                pauseScenario();
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid Command for ScenarioDispatcher!");
+        }
+    }
+
+    private void pauseScenario() {
+        //TODO: Implement
+    }
+
+    private void runScenario() {
+        Master master = Master.getInstance();
+        for (Event e : this.scenario.getEventList()) {
```

Comment:
This loop goes into the master, here we call master.runScenario(this.scenario)

Reply:
and it needs to be a separate class that is Runnable, then you would be able to run in a separate thread and pause it.

Reply:
this means that the master main function should also do the same, just create and run several instances of this new class.

Reply:
the configurator role in this case would be to predefine the outputs that would be connected to this input, for that you would need a scenario ID given in the configuration and the same ID used to run the scenario.

Reply:
Hey @alyhasansakr  , here when you say :
> and it needs to be a separate class that is Runnable, then you would be able to run in a separate thread and pause it.
What exactly do you mean? What should be a separate runnable class? The master?

Reply:
No, a new class, called MasterThread or something, it just contains this loop and runs in its own thread, and of course this needs to be generalized to the same loop in the original main.
In Scenario case, it would be created but not assigned an actual Scenario instance, until a Scenario is created, then the MasterThread instance should be started.